src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -239,7 +239,7 @@ public WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventTypes,
             int logicalPinNumber = GetLogicalPinNumber(pinNumber);
             if (!_openPins.Contains(logicalPinNumber))
             {
-                throw new InvalidOperationException("Can not wait for events from a pin that is not open.");
+                throw new InvalidOperationException($"Can not wait for events from pin {logicalPinNumber} because is not open.");
```

Comment:
```suggestion
                throw new InvalidOperationException($"Can not wait for events from pin {logicalPinNumber} because it is not open.");
```